src/main/kotlin/org/rust/lang/core/resolve/ref/RsPathReferenceImpl.kt

```diff
@@ -152,35 +168,102 @@ class RsPathReferenceImpl(
         return element.replace(elementNew)
     }
 
-    private object Resolver : (RsPath) -> List<BoundElementWithVisibility<RsElement>> {
-        override fun invoke(element: RsPath): List<BoundElementWithVisibility<RsElement>> {
-            return resolvePath(element)
+    private object Resolver : (RsElement) -> Any {
+        /**
+         * Returns `List<RsPathResolveResult<RsElement>>` if [root] is a single path or
+         * `Map<RsPath, List<RsPathResolveResult<RsElement>>>` if [root] contains nested paths.
+         */
+        override fun invoke(root: RsElement): Any {
+            val allPaths = root.collectNestedPathsFromRoot(root)
+            if (allPaths.isEmpty()) return emptyList<RsPathResolveResult<RsElement>>()
```

Comment:
Is it expected that `allPaths` will be empty if `root.parent is RsPathExpr` (such paths are resolved using inference cache)? If yes lets add a comment